src/common/dsp/effects/NimbusEffect.cpp

```diff
@@ -269,39 +270,39 @@ void NimbusEffect::init_ctrltypes()
             auto fx = &(p->storage->getPatch().fx[p->ctrlgroup_entry]);
             auto idx = p - fx->p;
 
-            static char res[TXT_SIZE];
-            res[0] = 0;
+            static std::string res;
             auto mode = fx->p[nmb_mode].val.i;
+
             switch (mode)
             {
             case 0:
                 if (idx == nmb_density)
-                    snprintf(res, TXT_SIZE, "%s", "Density");
+                    res = fmt::format("{:s}", "Density");
                 if (idx == nmb_texture)
-                    snprintf(res, TXT_SIZE, "%s", "Texture");
+                    res = fmt::format("{:s}", "Texture");
                 if (idx == nmb_size)
-                    snprintf(res, TXT_SIZE, "%s", "Size");
+                    res = fmt::format("{:s}", "Size");
                 break;
             case 1:
             case 2:
                 if (idx == nmb_density)
-                    snprintf(res, TXT_SIZE, "%s", "Diffusion");
+                    res = fmt::format("{:s}", "Diffusion");
                 if (idx == nmb_texture)
-                    snprintf(res, TXT_SIZE, "%s", "Filter");
+                    res = fmt::format("{:s}", "Filter");
                 if (idx == nmb_size)
-                    snprintf(res, TXT_SIZE, "%s", "Size");
+                    res = fmt::format("{:s}", "Size");
                 break;
             case 3:
                 if (idx == nmb_density)
-                    snprintf(res, TXT_SIZE, "%s", "Smear");
+                    res = fmt::format("{:s}", "Smear");
```

Comment:
This is just res=“smear” since strings copy properly

Reply:
Haha yeah, totally!